### PR TITLE
[DP-0000] Setup Delta-Io DeltaSharing Fork for AIQ

### PR DESCRIPTION
**Pull request by yannistze**
Initial Setup of Delta-Io DeltaSharing Fork for AIQ

## Test Notes

- `sbt "clean; compile"`
```
...
[success] created output: /Users/yannis.tzemos/delta-sharing/spark/target
...
[success] created output: /Users/yannis.tzemos/delta-sharing/server/target
...
[success] created output: /Users/yannis.tzemos/delta-sharing/client/target
...
[success] Total time: 12 s, completed Sep 5, 2024, 11:33:16 PM
```

- `sbt "clean; compile; publish"`
```
[info]  published delta-sharing-server_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-server_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-server_2.12-1.1.2-aiq0-yt2-SNAPSHOT.pom
[info]  published delta-sharing-server_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-server_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-server_2.12-1.1.2-aiq0-yt2-SNAPSHOT.jar
[info]  published delta-sharing-client_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-client_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-client_2.12-1.1.2-aiq0-yt2-SNAPSHOT.pom
[info]  published delta-sharing-client_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-client_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-client_2.12-1.1.2-aiq0-yt2-SNAPSHOT.jar
[info]  published delta-sharing-spark_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-spark_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-spark_2.12-1.1.2-aiq0-yt2-SNAPSHOT.pom
[info]  published delta-sharing-spark_2.12 to https://actioniq.jfrog.io/artifactory/aiq-sbt-local/io/delta/delta-sharing-spark_2.12/1.1.2-aiq0-yt2-SNAPSHOT/delta-sharing-spark_2.12-1.1.2-aiq0-yt2-SNAPSHOT.jar
[success] Total time: 11 s, completed Sep 5, 2024, 11:34:27 PM
```

Screenshot 2024-09-05 at 23 36 49

- `sbt test`
```
...
[info] - traffic goes through a proxy when a proxy configured *** FAILED ***
[info]   org.apache.http.conn.HttpHostConnectException: Connect to 172.16.24.51:52946 [/172.16.24.51] failed: Operation timed out
...
[info] Run completed in 5 minutes, 38 seconds.
[info] Total number of tests run: 14
[info] Suites: completed 8, aborted 0
[info] Tests: succeeded 14, failed 0, canceled 78, ignored 0, pending 0
[info] All tests passed.
[info] Run completed in 5 minutes, 38 seconds.
[info] Total number of tests run: 123
[info] Suites: completed 12, aborted 0
[info] Tests: succeeded 119, failed 4, canceled 41, ignored 0, pending 0
[info] *** 4 TESTS FAILED ***
[error] Failed tests:
[error]         io.delta.sharing.client.DeltaSharingFileSystemSuite
[info] Run completed in 5 minutes, 38 seconds.
[info] Total number of tests run: 60
[info] Suites: completed 10, aborted 0
[info] Tests: succeeded 60, failed 0, canceled 74, ignored 0, pending 0
[info] All tests passed.
[error] (client / Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 347 s (05:47), completed Sep 5, 2024, 11:42:58 PM
```

## Deploy Notes
- Merge to `master`
- `sbt "clean; compile; publish;"`
- Pickup in Flame and AIQ

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -214,20 +236,9 @@ lazy val releaseSettings = Seq(
...
-      Some("releases"  at nexus + "service/local/staging/deploy/maven2")
-    }
-  },
-
-  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
-
   releaseCrossBuild := true,
 
-  licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
+  licenses += ("Apache-2.0", url("http://opensource.org/licenses/Apache-2.0")),
```
Comment (MasterDDT) [inline build.sbt:241]: 🤷  idk if this matters

**Comment by MasterDDT** [comment]
So tests failed only bc no local delta sharing instance, rest were good on 17? 👍

File: build.sbt (hunk 1 above)
Comment (yannistze) [reply build.sbt:241]: Yeah, me neither 🤷‍♂️

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-version in ThisBuild := "1.1.2-SNAPSHOT"
+version in ThisBuild := "1.1.2-aiq0-SNAPSHOT"
```
Comment (MasterDDT) [inline version.sbt:1]: You can remove the `-SNAPSHOT`, we dont really need it...just make sure we always bump the aiq whenever we change, like usual

**Comment by yannistze** [comment]
> So tests failed only bc no local delta sharing instance, rest were good on 17? 👍

That's what I am seeing, yeah 🤔

**Review by MasterDDT — APPROVED** [review]
Some comments, else lgtm

File: build.sbt (hunk 1 above)
Comment (yannistze) [reply build.sbt]: Removed 👍

File: version.sbt (hunk 2 above)
Comment (yannistze) [reply version.sbt]: Yeap, done 👍

**Pull request merged**